source/include/azure_iot_adu_client.h

```diff
@@ -80,7 +80,6 @@ typedef struct AzureIoTADUClientDeviceProperties
  */
 typedef enum AzureIoTADUAction
 {
-    eAzureIoTADUActionDownload = 0,

```

Comment:
Is the numerical value important? If not, since this is beta and we don't provide ABI compat we should shift all other values up.

Reply:
We should also have an "unknown" action - i.e. an action that cannot be parsed.
This value could be assigned when parsing fails. Ideally, the state machine can still operate in those conditions by either ignoring or reporting the new action.

Reply:
I had thought about that. Just to align with what is in the docs, I will keep it as is for now and after we release, we can device to add an "unknown" one.

Reply:
Re the linkage - we should then keep the 0 value until it is removed from the spec (but fail accordingly).  It should have a deprecation API doc (similar to how we have in Embedded C SDK). We should also make sure we have all values in the documentation mirrored in our SDK.

Re the unknown: since the values are restricted by the protocol, it's likely better to have only those values. `unknown` will need to be signaled in a different way (e.g. via returned result during parsing).

Reply:
This isn't something we will do right now. If we put those in, it would be after public preview. For now, the sample will treat those deprecated values as "unknown actions"